packages/compass-query-bar/src/components/query-option.tsx

```diff
@@ -154,25 +173,30 @@ const UnthemedQueryOption: React.FunctionComponent<QueryOptionProps> = ({
             data-testid={`query-bar-option-${queryOption}-input`}
           />
         ) : (
-          <TextInput
-            aria-labelledby={`query-bar-option-input-${queryOption}-label`}
-            id={id}
-            data-testid="query-bar-option-input"
-            className={cx(
-              darkMode
-                ? numericTextInputDarkStyles
-                : numericTextInputLightStyles,
-              hasError && optionInputWithErrorStyles
+          <WithOptionDefinitionTextInputProps definition={optionDefinition}>
```

Comment:
Interesting, so TS was complaining about extra props? When looking at their docs I thought we can pass any props we want on top of the standard set. https://www.mongodb.design/component/text-input/documentation/

Reply:
> Interesting, so TS was complaining about extra props? 

Not that I am aware of – the comment above the definition of `WithOptionDefinitionTextInputProps` has some context around why the wrapper is being used rather than just passing the props through directly:

> ```js
> // Helper component to allow flexible computation of extra props for the TextInput
> // component if the query option definition suggests it. In particular,
> // using a separate component allows those extra props to use React hooks in their definition.
> ```

> When looking at their docs I thought we can pass any props we want on top of the standard set. https://www.mongodb.design/component/text-input/documentation/

I’m not sure but I’d read that page as saying that we can pass and props that are supported by the native HTML `input` element, not any props, but in any case, I don’t see why we would want to pass props that are not part of the standard set (and the type used here is `Partial<React.ComponentProps<typeof TextInput>>` to reflect that).